CMakeLists.txt

```diff
@@ -130,48 +130,16 @@ add_custom_command(
-        src/com/amazon/corretto/crypto/provider/AccessibleByteArrayOutputStream.java
-        src/com/amazon/corretto/crypto/provider/AesGcmSpi.java
-        src/com/amazon/corretto/crypto/provider/ConstantTime.java
-        src/com/amazon/corretto/crypto/provider/DebugFlag.java
-        src/com/amazon/corretto/crypto/provider/EcGen.java
-        src/com/amazon/corretto/crypto/provider/EvpHmac.java
-        src/com/amazon/corretto/crypto/provider/EvpKeyAgreement.java
-        src/com/amazon/corretto/crypto/provider/EvpKey.java
-        src/com/amazon/corretto/crypto/provider/EvpEcKey.java
-        src/com/amazon/corretto/crypto/provider/EvpEcPublicKey.java
-        src/com/amazon/corretto/crypto/provider/EvpEcPrivateKey.java
-        src/com/amazon/corretto/crypto/provider/EvpRsaKey.java
-        src/com/amazon/corretto/crypto/provider/EvpRsaPublicKey.java
-        src/com/amazon/corretto/crypto/provider/EvpRsaPrivateKey.java
-        src/com/amazon/corretto/crypto/provider/EvpRsaPrivateCrtKey.java
-        src/com/amazon/corretto/crypto/provider/EvpKeyFactory.java
-        src/com/amazon/corretto/crypto/provider/EvpKeyType.java
-        src/com/amazon/corretto/crypto/provider/EvpSignature.java
-        src/com/amazon/corretto/crypto/provider/EvpSignatureBase.java
-        src/com/amazon/corretto/crypto/provider/EvpSignatureRaw.java
-        src/com/amazon/corretto/crypto/provider/EcUtils.java
-        src/com/amazon/corretto/crypto/provider/ExtraCheck.java
-        src/com/amazon/corretto/crypto/provider/InputBuffer.java
-        src/com/amazon/corretto/crypto/provider/Janitor.java
-        src/com/amazon/corretto/crypto/provider/LibCryptoRng.java
-        src/com/amazon/corretto/crypto/provider/Loader.java
-        src/com/amazon/corretto/crypto/provider/AmazonCorrettoCryptoProvider.java
-        src/com/amazon/corretto/crypto/provider/NativeResource.java
-        src/com/amazon/corretto/crypto/provider/MiscInterfaces.java
-        src/com/amazon/corretto/crypto/provider/ReflectiveTools.java
-        src/com/amazon/corretto/crypto/provider/RsaCipher.java
-        src/com/amazon/corretto/crypto/provider/RsaGen.java
-        src/com/amazon/corretto/crypto/provider/RuntimeCryptoException.java
-        src/com/amazon/corretto/crypto/provider/SelfTestFailureException.java
-        src/com/amazon/corretto/crypto/provider/SelfTestResult.java
-        src/com/amazon/corretto/crypto/provider/SelfTestStatus.java
-        src/com/amazon/corretto/crypto/provider/SelfTestSuite.java
-        src/com/amazon/corretto/crypto/provider/ServiceProviderFactory.java
-        src/com/amazon/corretto/crypto/provider/Utils.java
-        ${GENERATED_JAVA_SRC}
-)
+# NOTE: CMake introduced the CONFIGURE_DEPENDS feature in 3.12, so condition
+#       its use on CMake version. If the feature is not available, you may need
+#       to `touch CMakeLists.txt` after adding/removing files for them to be
+#       detected by CMake.
+if (${CMAKE_VERSION} VERSION_LESS "3.12.0")
```

Comment:
Nit: If we already have an alternative for Cmake versions below `3.12.0`, is the second sentence necessary? same for the comment on Line 524

Reply:
i think so, my understanding is that without `CONFIGURE_DEPENDS`, cmake won't automatically detect new files. one way to force a re-build of the build system is to `touch CMakeLists.txt`. from [the docs](https://cmake.org/cmake/help/latest/command/file.html#filesystem):

> New in version 3.12: If the CONFIGURE_DEPENDS flag is specified, CMake will add logic to the main build system check target to rerun the flagged GLOB commands at build time. If any of the outputs change, CMake will regenerate the build system.

Reply:
Ah that's quite interesting, I misunderstood touching the file as "manually configuring it". Then I guess the comment's pretty helpful.